app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,320 @@
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.ListAdapter;
+import android.widget.TextView;
+
+import com.google.android.material.snackbar.Snackbar;
+import com.nextcloud.android.sso.AccountImporter;
+import com.nextcloud.android.sso.exceptions.AccountImportCancelledException;
+import com.nextcloud.android.sso.exceptions.AndroidGetAccountsPermissionNotGranted;
+import com.nextcloud.android.sso.exceptions.NextcloudFilesAppAccountNotFoundException;
+import com.nextcloud.android.sso.exceptions.NextcloudFilesAppNotInstalledException;
+import com.nextcloud.android.sso.exceptions.NoCurrentAccountSelectedException;
+import com.nextcloud.android.sso.helper.SingleAccountHelper;
+import com.nextcloud.android.sso.ui.UiExceptionManager;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.PreferenceActivity;
+import de.danoeh.antennapod.core.event.SyncServiceEvent;
+import de.danoeh.antennapod.core.preferences.GpodnetPreferences;
+import de.danoeh.antennapod.core.sync.SyncService;
+import de.danoeh.antennapod.dialog.AuthenticationDialog;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import static de.danoeh.antennapod.core.sync.SyncService.SYNC_PROVIDER_CHOICE_GPODDER_NET;
+import static de.danoeh.antennapod.core.sync.SyncService.SYNC_PROVIDER_CHOICE_NEXTCLOUD;
+
+public class SynchronizationPreferencesFragment extends PreferenceFragmentCompat {
+    private static final String PREFERENCE_SYNCHRONIZATION_DESCRIPTION = "preference_synchronization_description";
+    private static final String PREFERENCE_LOGIN = "pref_synchronization_authenticate";
+    private static final String PREF_GPODNET_SETLOGIN_INFORMATION = "pref_gpodnet_setlogin_information";
+    private static final String PREFERENCE_SYNC = "pref_synchronization_sync";
+    private static final String PREFERENCE_FORCE_FULL_SYNC = "pref_synchronization_force_full_sync";
+    private static final String PREFERENCE_LOGOUT = "pref_synchronization_logout";
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        addPreferencesFromResource(R.xml.preferences_synchronization);
+        setupScreen();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setTitle(R.string.synchronization);
+        updateScreen();
+        EventBus.getDefault().register(this);
+    }
+
+    private int getIconForSelectedSyncProvider(String currentSyncProviderKey) {
+        switch (currentSyncProviderKey) {
+            case SYNC_PROVIDER_CHOICE_GPODDER_NET:
+                return R.drawable.gpodder_icon;
+            case SYNC_PROVIDER_CHOICE_NEXTCLOUD:
+                return R.drawable.nextcloud_logo_svg;
+            default:
+                return R.drawable.ic_notification_sync;
```

Comment:
The notification icon is always white, so this doesn't work in the light theme. How about adding a new "cloud" icon for both here and on the main settings screen?

Reply:
i'd rather add a dark version for the light mode. How could i make android resolve to the correct file/image?

Reply:
Most icons (if they are not the ones used for notifications) already switch between light and dark mode automatically (xml contains an attr reference to the color). I think the icon should be consistent with the corresponding icon on the main settings screen - so please either use the one that is used there or add a new icon that is used in both locations. The circular arrows that you are currently using are already used for feed refresh, so I think it is confusing to use the same icon for sync, too.

Reply:
cloud icon it is